packages/flutter/test/rendering/paragraph_test.dart

```diff
@@ -222,6 +222,51 @@ void main() {
     expect(paragraph.debugNeedsPaint, isFalse);
   });
 
+  test('nested TextSpans in paragraph handle textScaleFactor correctly.', () {
+    final TextSpan testSpan = const TextSpan(
+      text: 'a',
+      style: const TextStyle(
+        fontSize: 10.0,
+      ),
+      children: const <TextSpan>[
+        const TextSpan(
+          text: 'b',
+          children: const <TextSpan>[
+            const TextSpan(text: 'c'),
+          ],
+          style: const TextStyle(
+            fontSize: 20.0,
+          ),
+        ),
+        const TextSpan(
+          text: 'd',
+        ),
+      ],
+    );
+    final RenderParagraph paragraph = new RenderParagraph(testSpan,
+        textDirection: TextDirection.ltr, textScaleFactor: 2.5);
+    paragraph.layout(const BoxConstraints());
+    expect(paragraph.size.height, equals(50.0));
+    expect(paragraph.size.width, equals(150.0));
+
+    // Test the sizes of nested spans.
+    final List<ui.TextBox> boxes = <ui.TextBox>[];
+    final String text = testSpan.toStringDeep();
+    for (int i = 0; i < text.length; ++i) {
```

Comment:
(I personally prefer `i += 1`, but that's not a Flutter style issue. Feel free to leave it as `++i`.)

Reply:
I learnt from Pascal, where it's spelt `Inc(Index);`. :-)

I'm averse to `++i` because it looks like an expression but is a statement with an effect. In general I prefer expressions to be side-effect free, because side-effects are a bug factory.